go/vt/wrangler/materializer.go

```diff
@@ -1305,3 +1312,28 @@ func (mz *materializer) forAllTargets(f func(*topo.ShardInfo) error) error {
 	wg.Wait()
 	return allErrors.AggrError(vterrors.Aggregate)
 }
+
+// checkIfTimeZoneIsLoaded is a light-weight consistency check to validate that, if a source time zone is specified to MoveTables,
+// that the current primary has the time zone loaded in order to run the convert_tz() function used by VReplication to do the
+// datetime conversions. We only check the current primaries on each shard and note here that it is possible a new primary
+// gets elected: in this case user will either see errors during vreplication or vdiff will report mismatches.
+func (mz *materializer) checkIfTimeZoneIsLoaded(ctx context.Context, tz string, keyspace string) error {
+	err := mz.forAllTargets(func(target *topo.ShardInfo) error {
+		targetPrimary, err := mz.wr.ts.GetTablet(ctx, target.PrimaryAlias)
+		if err != nil {
+			return vterrors.Wrapf(err, "GetTablet(%v) failed", target.PrimaryAlias)
+		}
+		query := fmt.Sprintf("select convert_tz(now(), %s, 'UTC')", encodeString(tz))
+		qrproto, err := mz.wr.tmc.ExecuteFetchAsApp(ctx, targetPrimary.Tablet, false, []byte(query), 1)
+		if err != nil {
+			return vterrors.Wrapf(err, "ExecuteFetchAsApp(%v, %s)", targetPrimary.Tablet, query)
+		}
+		qr := sqltypes.Proto3ToResult(qrproto)
+		gotDate := qr.Rows[0][0].ToString()
+		if gotDate == "" {
```

Comment:
I'm pretty sure (from looking at the `sqltypes.Value` code) that `gotDate` will be the string literal "NULL" here when we are not able to perform the conversion because either of the specified named time zones could not be found (either because one was invalid or the time zone tables have not been loaded). 

So I think that we should check for `gotDate == "NULL"` or **_even better_** to confirm we got a valid date and provide a more informative error message when we did not:
```
		dateFormat := "2006-01-02 15:04:05"
 		if gotDate, err := time.Parse(dateFormat, qr.Rows[0][0].ToString()); err != nil {
                       return fmt.Errorf("unable to perform time_zone conversions from %s to UTC — result of the attempt was: %s. Either the specified source time zone is invalid or the time zone tables have not been loaded on the %s tablet", tz, gotDate, targetPrimary.Alias)
 		}
```

Reply:
❤️